controllers/articlesController.js

```diff
@@ -311,4 +284,143 @@ export default class PostController {
+          };
+        });
+        const articlesCount = articles.length;
+        return res
+          .status(OK)
+          .json({ message: "Successful", articles, articlesCount });
+      }
+      return res.status(NOT_FOUND).json({
+        message: "No more articles found, still need to read more?"
+      });
+    } catch (error) {
+      return res
+        .status(INTERNAL_SERVER_ERROR)
+        .json({ message: "Can't get the articles, server error" });
+    }
+  }
+
+  /**
+   * @description It helps the user to update any of his articles.
+   * @param  {object} req - The request object
+   * @param  {object} res - The response object
+   * @returns {object} It returns the request's response object
+   */
+
+  static async update(req, res) {
+    try {
+      const { slug } = req.params;
+      const { user: { id: userId } = {}, body } = req;
+      const article = await Article.findOne({ where: { slug, userId } });
+      if (article) {
+        const allowed = _.pick(body, Object.keys(articleSchema));
+        const valid = await articleValidator(
+          allowed,
+          _.pick(articleSchema, Object.keys(allowed))
+        );
+        const { tagsList = [], ...rest } = valid;
+        if (tagsList.length) {
+          tagsList.map(async t => {
+            const tag = await Tag.findOrCreate({ where: { name: t } }).spread(
+              tg => tg
+            );
+            await article.addTagsList(tag);
+          });
+        }
+        const updated = await article.update(rest, {
+          where: { id: article.id, userId },
+          returning: true,
+          limit: 1
+        });
+        const tags = await updated
+          .getTagsList({ attributes: ["name"] })
+          .map(t => t.name || null);
+        return res.status(CREATED).json({
+          article: {
+            ...updated.get(),
+            author: _.pick(await updated.getAuthor(), ["username", "bio", "image"]),
+            tagsList: tags
+          },
+          message: `Updated successfully`
+        });
+      }
+      return res.status(UNAUTHORIZED).json({
+        message: `You can only update the article you authored`
+      });
+    } catch (error) {
+      return error.details
+        ? res.status(BAD_REQUEST).json({ message: error.details[0].message })
+        : res.status(INTERNAL_SERVER_ERROR).json({
+            message: "Can't update the article, server error"
+          });
+    }
+  }
+
+  /**
+   * @description It helps the user to delete any of his articles.
+   * @param  {object} req - The request object
+   * @param  {object} res - The response object
+   * @returns {object} It returns the request's response object
+   */
+
+  static async deleteOne(req, res) {
+    try {
+      const { slug } = req.params;
+      const { user: { id: userId } = {} } = req;
+      const article = await Article.findOne({ where: { slug } });
+      if (article.userId !== userId || !article) {
```

Comment:
`article.userId` may break if the article is `null`, which will raise server error. How about considering the scenario where a user is trying to delete an article which does not exist? What do you think about returning `404` specific error message, instead of `500`?